src/Core/Resolvers/AuthorizationPolicyHelpers.cs

```diff
@@ -46,23 +48,154 @@ public static class AuthorizationPolicyHelpers
+                                {
+                                    if (pathConfig.ColumnName is null || pathConfig.EntityName is null)
+                                    {
+                                        continue;
+                                    }
+                                    //Increment Table counter with the new JOIN so that we can have unique alias for each join
+                                    cosmosQueryStructure.TableCounter.Next();
+
+                                    fromClause = pathConfig.JoinStatement;
+                                    predicates = filterClause?.Expression.Accept(new ODataASTCosmosVisitor(pathConfig.Alias));
+
+                                    existQuery = GetExistQueryForCosmos(fromClause, predicates);
+                                }
+                                else
+                                {
+                                    predicates = filterClause?.Expression.Accept(new ODataASTCosmosVisitor($"{pathConfig.Path}.{pathConfig.ColumnName}"));
+                                }
+
+                                if (pathConfig.EntityName == entity.Key)
+                                {
+                                    if (!cosmosQueryStructure.DbPolicyPredicatesForOperations.TryGetValue(operationType, out string? _))
+                                    {
+                                        cosmosQueryStructure.DbPolicyPredicatesForOperations[operationType]
+                                                        = existQuery ?? predicates;
+                                    }
+                                    else
+                                    {
+                                        cosmosQueryStructure.DbPolicyPredicatesForOperations[operationType]
+                                                        += $" AND {existQuery ?? predicates}";
+                                    }
+                                }
+                            }
+                        });
+                }
+            }
+        }
+
+        /// <summary>
+        /// Generate Cosmos DB Query for the given fromClause and predicates.
+        /// </summary>
+        /// <param name="fromClause"></param>
+        /// <param name="predicates"></param>
+        /// <returns></returns>
+        private static string GetExistQueryForCosmos(string? fromClause, string? predicates)
+        {
+            string? existQuery = $"EXISTS " +
+                                $"(SELECT VALUE 1 " +
+                                    $"FROM {fromClause} ";
+            if (predicates is not null)
+            {
+                existQuery += $"WHERE {predicates})";
+            }
+            else
+            {
+                existQuery += ")";
+            }
+
+            return existQuery;
+        }
+
+        /// <summary>
+        /// It process the filter clauses for the given elemental operation.
+        /// </summary>
```

Comment:
Can you elaborate on the side effects of "processing the filter"?
- which passed in args are modified (side effects), if any
- example of what a returned filter clause list may look like?

Reply:
1.nothing is modified, it returns OData filter clause.
2. I am not sure what example, you are looking for, it totally depends on the postProcessCallback